client/client.go

```diff
@@ -0,0 +1,402 @@
+/*******************************************************************************
+ * Copyright (c) 2021, 2025 Genome Research Ltd.
+ *
+ * Author: Sendu Bala <sb10@sanger.ac.uk>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/VertebrateResequencing/wr/jobqueue"
+	jqs "github.com/VertebrateResequencing/wr/jobqueue/scheduler"
+	"github.com/inconshreveable/log15"
+	"github.com/rs/xid"
+	"github.com/wtsi-ssg/wr/clog"
+)
+
+type Error string
+
+func (e Error) Error() string { return string(e) }
+
+const errDupJobs = Error("some of the added jobs were duplicates")
+
+// PretendSubmissions as a non-empty string causes SubmitJobs to only record the
+// jobs for retrieval by SubmittedJobs(); no wr manager server is needed or
+// used.
+//
+// If set to a number, SubmitJobs will print JSON encoded data to that file
+// descriptor.
```

Comment:
Can you add some example code for how you'd set this when using this pkg from another repo?